mlflow/utils/autologging_utils/__init__.py

```diff
@@ -486,17 +489,38 @@ def autologging_is_disabled(integration_name):
 
 
 @contextlib.contextmanager
-def disable_autologging():
+def disable_autologging(exceptions=None):
     """
     Context manager that temporarily disables autologging globally for all integrations upon
     entry and restores the previous autologging configuration upon exit.
+
+    Args:
+        exceptions: flavors that we do not disable
     """
+    if exceptions is None:
+        exceptions = []
     global _AUTOLOGGING_GLOBALLY_DISABLED
+    global _AUTOLOGGING_GLOBALLY_DISABLED_EXCEPTIONS
     _AUTOLOGGING_GLOBALLY_DISABLED = True
+    _AUTOLOGGING_GLOBALLY_DISABLED_EXCEPTIONS = exceptions
     try:
         yield None
     finally:
         _AUTOLOGGING_GLOBALLY_DISABLED = False
+        _AUTOLOGGING_GLOBALLY_DISABLED_EXCEPTIONS = []
+
+
+@contextlib.contextmanager
+def restrict_langchain_autologging_to_traces_only():
+    if importlib.util.find_spec("langchain") is None:
+        yield None
+    else:
+        prev_langchain_params = AUTOLOGGING_INTEGRATIONS.get(mlflow.langchain.FLAVOR_NAME)
+        mlflow.langchain.autolog(log_inputs_outputs=False)
```

Comment:
@serena-ruan Shouldn't we have a `log_traces` flag and remove the `log_inputs_outputs` flag since we no longer need to log inputs & outputs as a DataFrame?

@jessechancy While we're here, perhaps we can make that change?

Reply:
Can we do this in a followup PR? There are some discussions related to how this can be backwards compatible, and what to gate behind which flag / default values that might block this PR